test/close.test.js

```diff
@@ -2,83 +2,86 @@
 
 const net = require('node:net')
 const http = require('node:http')
-const { test } = require('tap')
+const { test } = require('node:test')
 const Fastify = require('..')
 const { Client } = require('undici')
 const split = require('split2')
 const { sleep } = require('./helper')
 
-test('close callback', t => {
+test('close callback', (t, testDone) => {
   t.plan(7)
   const fastify = Fastify()
   fastify.addHook('onClose', onClose)
   function onClose (instance, done) {
-    t.type(fastify, this)
-    t.type(fastify, instance)
-    t.equal(fastify, this)
-    t.equal(fastify, instance)
+    t.assert.ok(typeof fastify === typeof this)
+    t.assert.ok(typeof fastify === typeof instance)
+    t.assert.strictEqual(fastify, this)
+    t.assert.strictEqual(fastify, instance)
     done()
+    testDone()
   }
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
+    t.assert.ifError(err)
 
     fastify.close((err) => {
-      t.error(err)
-      t.ok('close callback')
+      t.assert.ifError(err)
+      t.assert.ok('close callback')
```

Comment:
shouldn't the `testDone` call be here instead?

Reply:
yes, you're right!